src/hoc/withTheme.js

```diff
@@ -32,15 +33,14 @@ const wrapWithTheme = (Component: ReactClass<any>) => {
 
     componentWillMount() {
       const styledContext = this.context[CHANNEL_NEXT]
-      if (styledContext === undefined) {
+      if (!styledContext && !this.constructor.defaultProps) {
         // eslint-disable-next-line no-console
-        console.error('[withTheme] Please use ThemeProvider to be able to use withTheme')
-        return
+        console.warn('[withTheme] You are not using a ThemeProvider nor passing a theme in defaultProps')
       }
```

Comment:
Needs a case here for when `if (!styledContext && this.contructor.defaultProps)` to use the defaultProps or props I think.

Reply:
What should that case do then? Just a `setState` with the theme from `defaultProps`?

Reply:
Should probably still use `determineTheme` with the props, with undefined as the context theme?

Reply:
Ok, thank you! 👌